### PR TITLE
[Mission] Mise en place de l'état de complétion des données des actions

### LINKED ISSUE #3037: [Mission] Mise en place de l'état de complétion des données des actions
### Liste 

- Ajout d'un filtre sur statut & complétion des données 👍🏻 
- Ajouter deux colonnes statut et complétion des données 👍🏻 
  - Si un état est non-complété parmi la mission et les actions, on passe à "à compléter" 👍🏻 
- Rajouter les filtre "Missions avec actions Fish" (voir ticket #3008) 👍🏻 
- Supprimer le filtre "Ouvert par" (voir ticket #3008) 👍🏻 

#### Mission
- Afficher les tags de complétion 👍🏻 
- A chaque enregistrement, calculer l'état de complétion de la mission et de l'action pour l'enregistrer en BDD 👍🏻 
- Afficher le bandeau quand on revient sur une mission terminée mais non-complétée

#### Action

- A chaque enregistrement, calculer l'état de complétion de la mission et de l'action pour l'enregistrer en BDD 👍🏻 
- En dessous du titre de l'action, afficher les champs à compléter, ex. "12 champs à compléter pour la stat" 👍🏻 

### Données

- Rajouter le champ `completion` dans les actions
- Si le trigramme de clôture est présent dans les actions, on migre les `isClosed` en `COMPLETED`

### PR DESCRIPTION
**Pull request by louptheron**
## Linked issues

- Resolve #3037

----

- [ ] Tests E2E (Cypress)

## Summary by CodeRabbit

- **New Features**
	- Introduced a completion status for mission actions, allowing users to mark actions as either completed or to be completed.
	- Added new UI components to visually indicate the completion status of mission actions and the presence of missing required fields.
	- Implemented a custom hook to calculate missing fields for mission action forms and the overall completion status of missions.
- **Enhancements**
	- Updated mission action forms to require a completion status, enhancing data integrity and user experience.
	- Improved the visual representation of mission statuses with new icons and tags, facilitating a more intuitive interface.
- **Bug Fixes**
	- Corrected text representation for unknown vessels in action cards.
- **Documentation**
	- Added comprehensive tests to ensure reliability and accuracy of the new completion status feature and related components.

### CODE REVIEW

File: frontend/cypress/e2e/side_window/mission_form/air_control.spec.ts [hunk 1]
```diff
@@ -55,9 +58,8 @@ context('Side Window > Mission Form > Air Control', () => {
     // Saisi par
     cy.fill('Saisi par', 'Marlin')
 
-    // Clôturé par
-    // TODO Handle multiple inputs with same label via an `index` in monitor-ui.
-    cy.get('[name="closedBy"]').eq(1).type('Alice')
+    // Complété par
+    cy.fill('Complété par', 'Alice')
```
Comment (claire2212) [inline frontend/cypress/e2e/side_window/mission_form/air_control.spec.ts:62]: c'est pas un trigramme sur Fish?

File: frontend/cypress/e2e/side_window/mission_form/main_form.spec.ts [hunk 2]
```diff
@@ -676,7 +542,7 @@ context('Side Window > Mission Form > Main Form', () => {
                 }
               ],
               // MODIFIED FIELD
-              endDateTimeUtc: '2024-02-13T09:49:40.350661Z',
+              endDateTimeUtc: '2070-02-13T09:49:40.350661Z',
```
Comment (claire2212) [inline frontend/cypress/e2e/side_window/mission_form/main_form.spec.ts:545]: ah ouais, carrément!

File: frontend/src/features/Mission/__tests__/getMissionCompletionStatus.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      CompletionStatus.TO_COMPLETE
+    ])
+    expect(missionCompletionToComplete).toBe(FrontCompletionStatus.TO_COMPLETE)
+
+    const missionCompletionToCompleteEnded = getMissionCompletionFrontStatus(DUMMY_MISSION_DONE, [
+      CompletionStatus.TO_COMPLETE
+    ])
+    expect(missionCompletionToCompleteEnded).toBe(FrontCompletionStatus.TO_COMPLETE_MISSION_ENDED)
+
+    const missionCompletionUpcoming = getMissionCompletionFrontStatus(DUMMY_MISSION_UPCOMING, [
```
Comment (claire2212) [inline frontend/src/features/Mission/__tests__/getMissionCompletionStatus.test.ts:33]: normalement pour une mission à venir il n'y a pas de statut de complétion il me semble, non?

File: frontend/src/features/Mission/components/MissionForm/ActionList/ActionCard.tsx [hunk 4]
```diff
@@ -19,52 +19,34 @@ type ActionCardProps = Readonly<{
...
           <DateLabel title={missionAction.actionDatetimeUtc}>
             <b>{formatDateLabel(startDateAsDayjs.format('DD MMM'))}</b> à {startDateAsDayjs.format('HH:mm')}
+            {missionAction.source === Mission.MissionSource.MONITORFISH && (
+              <CompletionStatusIcon missionAction={missionAction as MissionActionForTimeline} />
+            )}
           </DateLabel>
         )}
 
         <InnerWrapper
-          $isOpen={isOpen}
```
Comment (claire2212) [inline frontend/src/features/Mission/components/MissionForm/ActionList/ActionCard.tsx:51]: tu n'es met plus de bordure bleue sur l'action qui est ouverte?

File: frontend/src/features/Mission/components/MissionForm/utils/validateMissionForms.ts [hunk 5]
```diff
@@ -33,7 +33,7 @@ export function validateMissionForms(
           ...actionFormValues,
           // There is no closure validation schema for the air control action form
           isValid: isClosureValidation
```
Comment (claire2212) [inline frontend/src/features/Mission/components/MissionForm/utils/validateMissionForms.ts:35]: `isCompletionValidation`?

File: frontend/src/features/Mission/index.ts [hunk 6]
```diff
@@ -59,22 +60,25 @@ export const getMissionFeaturePoint = ({
...
     isDone: booleanToInt(missionStatus === MissionStatus.DONE),
     isInProgress: booleanToInt(missionStatus === MissionStatus.IN_PROGRESS),
     isLandMission: mission.missionTypes.length === 1 && mission.missionTypes.includes(MissionType.LAND),
     isMultiMission: mission.missionTypes.length > 1,
     isSeaMission: mission.missionTypes.length === 1 && mission.missionTypes.includes(MissionType.SEA),
     isUpcoming: booleanToInt(missionStatus === MissionStatus.UPCOMING),
+    missionCompletion: getMissionCompletionFrontStatus(mission, actionsCompletion),
     missionId: mission.id,
-    missionSource: mission.missionSource,
+    missionSources: getMissionSources(actions, envActions),
```
Comment (claire2212) [inline frontend/src/features/Mission/index.ts:81]: encore utile?

File: frontend/src/features/Mission/index.ts [hunk 7]
```diff
@@ -183,3 +187,22 @@ export function getMissionSourceTagText(missionSource: MissionSource | undefined
       return 'Origine inconnue'
   }
 }
+
```
Comment (claire2212) [inline frontend/src/features/Mission/index.ts:190]: la fonction au-dessus `getMissionSourceTagText` est toujours utilisée?

File: frontend/cypress/e2e/side_window/mission_form/air_control.spec.ts (hunk 1 above)
Comment (louptheron) [reply frontend/cypress/e2e/side_window/mission_form/air_control.spec.ts:62]: Si, mais pas les tests... On n'a pas de validation sur ce champ

File: frontend/src/features/Mission/components/MissionForm/MainForm/index.tsx [hunk 8]
```diff
@@ -117,8 +117,8 @@ function UnmemoizedMainForm({ initialValues, missionId, onChange }: MainFormProp
               </RelatedFieldGroupWrapper>
 
               <InlineFieldGroupWrapper>
-                <FormikTextInput isRequired label="Ouvert par" name="openBy" />
-                <FormikTextInput isRequired label="Clôturé par" name="closedBy" />
+                <FormikTextInput label="Ouvert par" name="openBy" />
+                <FormikTextInput label="Clôturé par" name="closedBy" />
```
Comment (claire2212) [inline frontend/src/features/Mission/components/MissionForm/MainForm/index.tsx:121]: `label="Complété par"` ?

**Review by claire2212 — APPROVED** [review]

File: frontend/src/features/Mission/__tests__/getMissionCompletionStatus.test.ts (hunk 3 above)
Comment (louptheron) [reply frontend/src/features/Mission/__tests__/getMissionCompletionStatus.test.ts:33]: Ah bien vu

File: frontend/src/features/Mission/components/MissionForm/ActionList/ActionCard.tsx (hunk 4 above)
Comment (louptheron) [reply frontend/src/features/Mission/components/MissionForm/ActionList/ActionCard.tsx:51]: Non, car on n'affiche plus les contrôles "en cours" (qui n'ont pas de trigramme de clôture)

File: frontend/src/features/Mission/index.ts (hunk 6 above)
Comment (louptheron) [reply frontend/src/features/Mission/index.ts:81]: Bien vu!

File: frontend/src/features/Mission/components/MissionForm/MainForm/index.tsx (hunk 8 above)
Comment (louptheron) [reply frontend/src/features/Mission/components/MissionForm/MainForm/index.tsx:121]: En effet!

**Pull request merged**